Gp/general refactoring

JackPiri opened the pull request:
Very general refactoring.

a-petrini commented on src/test/util_tests.cpp:
```diff
@@ -414,21 +414,22 @@ BOOST_AUTO_TEST_CASE(test_FormatSubVersion)
-    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments2), std::string("/Test:0.9.99-alpha(comment1; comment2)/"));
-    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99950, comments2), std::string("/Test:0.9.99(comment1; comment2)/"));
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments2), std::string("/Test:0.9.99-beta1(comment1; comment2)/"));
+    BOOST_CHECK_EQUAL(FormatSubVersion("Test", 99900, comments2), std::string("/Test:0.9.99-beta1(comment1; comment2)/"));
```
Just a minor non-blocking comment: after refactoring, lines 427 and 428 appear to execute the same check

JackPiri replied:
Yes sure, this is just duplication. Modified accordingly.

a-petrini replied:
Thanks!

a-petrini reviewed (approved):
Cool PR, thanks! Left a minor non-blocking comment, feel free to ignore!

JackPiri committed:
Very general refactoring

syntax and style cleanup
mapArgs usage
client versioning alignment
fixed misplaced semicolons in python tests
python tests cleaning
